### PR TITLE
[ADD] report_qr

### PR DESCRIPTION
**Pull request by etobella**
This PR replaces OCA/web#1229

### CODE REVIEW

**Comment by etobella** [comment]
ping @hveficent @jbeficent

**Review by HviorForgeFlow — APPROVED** [review]

**Review by jarroyomorales — APPROVED** [review]
Increase coverage if possible

**Comment by etobella** [comment]
@jarroyomorales Added some tests

**Comment by HviorForgeFlow** [comment]
test name seems no to be related with tested functionality in test. Maybe some description would be fine

**Comment by etobella** [comment]
@hveficent splitted into two tests with better names (I hope :wink: )

**Comment by HviorForgeFlow** [comment]
Much better

**Comment by etobella** [comment]
@jbeficent This PR was approved by you in another PR, It is still right for you?

**Comment by jarroyomorales** [comment]
Already green :green_apple:, @jbeficent Can you review?

**Comment by rafaelbn** [comment]
@OCA/reporting-engine-maintainers this PR is waiting a PSC :smile:

**Review by rafaelbn — APPROVED** [review]

File: report_qr/models/ir_actions_report.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    @api.model
+    def qr_generate(self, value, box_size=3, border=5, **kwargs):
+        try:
+            qr = qrcode.make(value, box_size=box_size, border=border, **kwargs)
+            arr = io.BytesIO()
+            qr.save(arr, format='png')
```
Comment (chienandalu) [inline report_qr/models/ir_actions_report.py]: Won't it save to png as default? I'd leave it open so an image_factory (like svg) can be passed to the controller

Comment (pedrobaeza) [reply report_qr/models/ir_actions_report.py]: Please see https://github.com/OCA/reporting-engine/pull/324/commits/3c9b5d589eb0bcfafcd910651295eb1cbdde25ea#diff-0dcdf5095cb728e790b48d4cea6d0ed4R16 for better extension

**Comment by etobella** [comment]
@chienandalu I cherry picked your commit

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**